### PR TITLE
fix(lint): Enable eslint for system-addon and its export to mozilla-central

### LINKED ISSUE #2229: Enable eslint for system-addon/
Right now we don't have eslint targeting `.jsm`s. There are a few issues we would need to sort out (like `no-unused-vars` conflicting with jsm-style exports, for example) to get this to work

### PR DESCRIPTION
**Pull request by Mardak**
Fix partially #2229 for this repository `eslint`ing and while exporting to `mozilla-central` where the top-level `.eslintrc.js` is only used for this repository and falling back to whatever is used for `browser/` (currently, generally less strict than here). I switched to the `eslintrc` `.js` format instead of `json` to match.

Is the approach of  `this.ActivityStream = class ActivityStream { ` acceptable? I suppose the class could be anonymous at that point? Alternatively is the `/* exports */` pattern.

Linting was partially failing on pine because the dot files weren't being copied over. Four rules were enabled to match mozilla-central and were autofixed except for `no-shadow`.

r? @k88hudson

### CODE REVIEW

File: yamscripts.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ scripts:
 
 # lint: Run eslint and sass-lint
   lint:
-    eslint: esw --ext=.js,.jsx,.json . .storybook
+    eslint: esw --ext=.js,.jsm,.json,.jsx . .storybook
```
Comment (k88hudson) [inline yamscripts.yml]: I don't actually think we actually use any `.jsx` files

File: .eslintignore [hunk 2]
```diff
@@ -9,5 +8,3 @@ firefox/
 logs/
 stats.json
 test/test-ColorAnalyzer.js
-test/test-SimpleStorage.js
-system-addon/
```
Comment (k88hudson) [inline .eslintignore]: Should we maybe add `system-addon/vendor/` here?

File: yamscripts.yml (hunk 1 above)
Comment (Mardak) [inline yamscripts.yml]: Removed `.jsx`

File: .eslintignore (hunk 2 above)
Comment (Mardak) [inline .eslintignore]: Added

**Comment by as-pine-proxy** [comment]
[Treeherder: e331e030d82a60746740141a45ff893a03890295](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=e331e030d82a60746740141a45ff893a03890295)

**Comment by dmose** [comment]
From the pine treeherder run, it looks like it's trying to lint the bundle itself.  I wonder how hard it would be to make eslint errs for the bootstrapped add-on show up in travis too, so that we'd catch them before merging...

**Pull request merged**